daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1310,9 +1326,16 @@ private[lf] object SBuiltin {
       methodName: MethodName,
   ) extends SBuiltin(1) {
     override private[speedy] def execute(args: util.ArrayList[SValue], machine: Machine): Unit = {
-      val (tyCon, record) = getSAnyContract(args, 0)
-      machine.ctrl =
-        SEApp(SEVal(ImplementsMethodDefRef(tyCon, ifaceId, methodName)), Array(SEValue(record)))
+      val (templateId, record) = getSAnyContract(args, 0)
```

Comment:
Assuming the previous code was not broken (I don’t think it was), we seem to have the invariant that an interface value is always an actual interface value (contrary to contract ids). So the error here would be an internal error not something users can run into.